File: fibonacci1.c
#include <stdio.h>
#include <gmp.h>
#include <time.h>

void fib(mpz_t rez, int n) {

	// ustavitvena pogoja za trivialna primera
	if(n == 0) {
		mpz_set_ui(rez, 0);
		return;
	}
	else if(n == 1) {
		mpz_set_ui(rez, 1);
		return;
	}
	// izracun preprostejsega primera fib(n-1) + fib(n-2)
	else {
		fib(rez, n-1);
		mpz_t a;
		mpz_init(a);
		mpz_set(a, rez);
		
		fib(rez, n-2);
		mpz_t b;
		mpz_init(b);
		mpz_set(b, rez);

		mpz_add(rez, a, b);
		mpz_clear(a);
		mpz_clear(b);
	}

}

int main() {

	// spremenljivka n za izracun fib(n)
	int n = 10;

	struct timespec startTime, endTime;
	// zacetni cas merjenja
	clock_gettime(CLOCK_MONOTONIC_RAW, &startTime);

	mpz_t rez;
	mpz_init(rez);
	fib(rez, n);

	// koncni cas merjenja
	clock_gettime(CLOCK_MONOTONIC_RAW, &endTime);
	// izracun casa merjenja
	int timeElapsed = (endTime.tv_sec - startTime.tv_sec) * 1000 + (endTime.tv_nsec - startTime.tv_nsec) / 1000000;

	// izpis rezultatov
	mpz_out_str(stdout, 10, rez);
	printf("\n");
	printf("cas: %d ms\n", timeElapsed);

	mpz_clear(rez);

	return 0;
}

File: fibonacci1.java
import java.math.BigInteger;

public class fibonacci1 {
	
	static BigInteger fib(int n) {
		
		// ustavitvena pogoja za trivialna primera
		if(n == 0) {
			return new BigInteger("0");
		}
		else if(n == 1) {
			return new BigInteger("1");
		}
		// izracun preprostejsega primera fib(n-1) + fib(n-2)
		else {
			BigInteger b = fib(n-1).add(fib(n-2));
			return b;
		}
		
	}
	
	public static void main(String[] args) {
		
		// spremenljivka n za izracun fib(n)
		int n = 10;
		
		// zacetni cas merjenja
		long startTime = System.currentTimeMillis();
		
		BigInteger rez = fib(n);
		
		// koncni cas merjenja
		long endTime = System.currentTimeMillis();
		// izracun casa merjenja
		long timeElapsed = endTime - startTime;
		
		// izpis rezultatov
		System.out.println(rez);
		System.out.println("cas: "+timeElapsed+" ms");
		
	}
	
}